homeassistant/components/tile/__init__.py

```diff
@@ -33,6 +33,31 @@ async def async_setup_entry(hass, entry):
     hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id] = {}
     hass.data[DOMAIN][DATA_TILE][entry.entry_id] = {}
 
+    # The existence of shared Tiles across multiple accounts requires an entity ID
+    # change:
+    #
+    # Old: tile_{uuid}
+    # New: {username}_{uuid}
+    #
+    # Find any entities with the old format and update them:
+    ent_reg = entity_registry.async_get(hass)
+    for entity in [
+        e
+        for e in ent_reg.entities.values()
+        if e.platform == DOMAIN
+        and not e.unique_id.startswith(f"{DOMAIN}_{entry.data[CONF_USERNAME]}")
```

Comment:
I think `{DOMAIN}_` is not nnecessary here.

Reply:
But that will rewrite unique IDs for any entity in _any_ domain that don't match the current format, right?

Reply:
New `unique_id` doesn't contain `DOMAIN` so I think this line should look like this
```
and not e.unique_id.startswith(entry.data[CONF_USERNAME])
```
Am I right?

Reply:
You already filter with `e.platform == DOMAIN` for that.

Reply:
Sorry, @bieniu, I misread your original comment. Got it now. 👍🏻